### DISCUSSION TITLE
Top showing multiple php-fpm processes

### DISCUSSION THREAD

**Discussion by rbnoorman** [nextcloud/all-in-one#1282] [Questions] [answered]
Hi all, 

I run a Ubuntu server VM in Proxmox with portainer and Nextcloud-AIO. I noticed that this server was using a lot of memory so I looked into what could cause it. I noticed there are several processess running by user www-data called php-fpm, these continually require quite a bit of memory. 

Is this normal? I don't have any other services running on this VM.

If normal, can someone explain what these processes do? Are they outside of docker, or does the 'top' command also show procesess which are running inside of Docker containers?

Thanks for your response!

**Comment by szaimen** [comment, marked as answer]
> If normal, can someone explain what these processes do? 

They are normal and these are the php processes that run Nextcloud.

> Are they outside of docker, or does the 'top' command also show procesess which are running inside of Docker containers?

I guess it makes sense for you to read about how docker works but yes, this is indeed the normal behaviour.
